parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoMessageConverter.java

```diff
@@ -132,23 +138,24 @@ public void add(Object value) {
 
     LogicalTypeAnnotation logicalTypeAnnotation = parquetType.getLogicalTypeAnnotation();
     if (logicalTypeAnnotation == null) {
-      return newScalarConverter(parent, parentBuilder, fieldDescriptor, parquetType);
+      return newScalarConverter(parent, parentBuilder, fieldDescriptor, parquetType, specCompliant);
     }
 
     return logicalTypeAnnotation.accept(new LogicalTypeAnnotation.LogicalTypeAnnotationVisitor<Converter>() {
       @Override
       public Optional<Converter> visit(LogicalTypeAnnotation.ListLogicalTypeAnnotation listLogicalType) {
-        return of(new ListConverter(parentBuilder, fieldDescriptor, parquetType));
+        return of(new ListConverter(parentBuilder, fieldDescriptor, parquetType, specCompliant));
       }
 
       @Override
       public Optional<Converter> visit(LogicalTypeAnnotation.MapLogicalTypeAnnotation mapLogicalType) {
-        return of(new MapConverter(parentBuilder, fieldDescriptor, parquetType));
+        return of(new MapConverter(parentBuilder, fieldDescriptor, parquetType, specCompliant));
       }
-    }).orElse(newScalarConverter(parent, parentBuilder, fieldDescriptor, parquetType));
+    }).orElse(newScalarConverter(parent, parentBuilder, fieldDescriptor, parquetType, specCompliant));
   }
```

Comment:
In fact, the root cause is we use orElse here, instead of orElseGet.
orElseGet is the lazy way to get the fallback value, while orElse is not. In the case of repeated fields (list and map), it was still trying to call newScalarConverter while using orELse, which is irrelevant (and thus the bug).

Reply:
Thanks @qinghui-xu , looks like the fix you suggested works well!

In this case, the good news: this is a bug, which is not yet released, because that orElse was added by me in PARQUET-1410 (my bad). Bad news: it is a regression, would be nice to get this committed before next release.

Reply:
Thanks for the prompt fix.